Description: read policies from a snap-aware path if it exists
 (typically /var/snap/chromium/current/policies),
 and fall back to the path used by the chromium-browser deb package if not
Author: Olivier Tilloy <olivier.tilloy@canonical.com>
Bug: https://bugs.launchpad.net/ubuntu/+source/chromium-browser/+bug/1714244
Bug: https://bugs.launchpad.net/ubuntu/+source/chromium-browser/+bug/1866732
Forwarded: not-needed

--- a/chrome/common/chrome_paths.cc
+++ b/chrome/common/chrome_paths.cc
@@ -4,4 +4,5 @@
 #include "chrome/common/chrome_paths.h"
 
+#include "base/environment.h"
 #include "base/files/file_path.h"
 #include "base/files/file_util.h"
@@ -423,7 +424,14 @@ bool PathProvider(int key, base::FilePat
       break;
 #if BUILDFLAG(IS_POSIX) && !BUILDFLAG(IS_MAC) && !BUILDFLAG(IS_OPENBSD)
     case chrome::DIR_POLICY_FILES: {
-      cur = base::FilePath(policy::kPolicyPath);
+      std::unique_ptr<base::Environment> env(base::Environment::Create());
+      std::string snap_data;
+      if (env->GetVar("SNAP_DATA", &snap_data)) {
+        cur = base::FilePath(snap_data);
+        cur = cur.Append(FILE_PATH_LITERAL("policies"));
+        if (base::PathExists(cur)) break;
+      }
+      cur = base::FilePath(FILE_PATH_LITERAL("/etc/chromium-browser/policies"));
       break;
     }
 #endif
